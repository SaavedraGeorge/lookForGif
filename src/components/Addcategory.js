import { useState } from "react";
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
    // { setCategories } es la destructuracion del props
    const [inputValue, setinputValue] = useState('');

    const handleInputChange = (e) => {
        setinputValue(e.target.value);
    };

    const handleSubmit = (e) => {
        e.preventDefault();
        //previene el comportamiento defoult de submit, no actualiza la pagina completa
        if (inputValue.trim().length > 2) {
            // trim() borra los espacios antes y despues
            setCategories(categorias => [inputValue, ...categorias]);
            setinputValue('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input
                type='text'
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    )
}
AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}