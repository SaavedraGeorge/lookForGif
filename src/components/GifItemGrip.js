import React from 'react';
import PropTypes from 'prop-types';

export const GifItemGrip = ({id, title, url}) => {
    return (
        <div className="card animate__animated animate__fadeIn">
            <p>{title}</p>
            <img src={url} alt={title} />
        </div>
    )
}
GifItemGrip.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}