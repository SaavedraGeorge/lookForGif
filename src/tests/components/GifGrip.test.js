import React from 'react';
import { shallow } from 'enzyme';
import { GifGrip } from '../../components/GifGrip'

describe('Pruebas en AddCategory', () => {

    const category = 'los simpson';

    test('should show <GifGrip /> whith 0 defoult', () => {
        const wrapper = shallow(<GifGrip category={ category } />);
        expect(wrapper).toMatchSnapshot()
    });


})