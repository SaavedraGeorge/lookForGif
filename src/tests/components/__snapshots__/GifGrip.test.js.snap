// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en AddCategory should show <GifGrip /> whith 0 defoult 1`] = `
<Fragment>
  <h2>
    los simpson
  </h2>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grip"
  />
</Fragment>
`;
